src/main/java/org/mongojack/JacksonMongoCollection.java

```diff
@@ -334,54 +335,18 @@ public UpdateResult updateMany(DBQuery.Query query, DBUpdate.Builder update, boo
     }
 
     /**
-     * Performs an update operation.
-     * 
-     * @param query
-     *            search query for old object to update
-     * @param object
-     *            object with which to update <tt>query</tt>
-     * @param upsert
-     *            if the database should create the element if it does not exist
-     * @param concern
-     *            the write concern
-     * @return The write result
-     * @throws MongoWriteException
-     *             If the write failed due some other failure specific to the update command
-     * @throws MongoWriteConcernException
-     *             If the write failed due being unable to fulfill the write concern
-     * @throws MongoException
-     *             If an error occurred
+     * @deprecated Use {@link #replaceOne(DBQuery.Query, Object)} instead.
      */
+    @Deprecated
     public UpdateResult update(DBQuery.Query query, T object, boolean upsert, WriteConcern concern) throws MongoException, MongoWriteException,
             MongoWriteConcernException {
-        if (concern != null) {
-            return mongoCollection.withWriteConcern(concern).updateOne(serializeQuery(query), convertToDocument(object), new UpdateOptions().upsert(
-                    upsert));
-        } else {
-            return mongoCollection.updateOne(serializeQuery(query), convertToDocument(object), new UpdateOptions().upsert(upsert));
-        }
-
+        return replaceOne(query, object, upsert, concern);
```

Comment:
Let's not change the implementation of this method. It would be dangerous for anyone relying on the previous behavior

I'm fine with deprecating it and providing a new `replaceOne` method though